### PR TITLE
CM-71: added support for applying queryset based on custom filters provided by frontend 

### PR DESCRIPTION
**Pull request by sniedzielski**
part of ticket: https://openimis.atlassian.net/browse/CM-71

- added additional implementation of interface for applying custom filters provided by frontend into Django queryset
- additional methods to clean input for filters and for casting from string (frontend provides data for filters values as strings)

### CODE REVIEW

File: social_protection/custom_filters.py [hunk 1]
```diff
@@ -44,6 +46,27 @@ def load_definition(self, tuple_type: type, **kwargs) -> List[namedtuple]:
         list_of_tuple_with_definitions = self.__process_schema_and_build_tuple(benefit_plan, tuple_type)
         return list_of_tuple_with_definitions
 
+    def apply_filter_to_queryset(self, custom_filters: List[namedtuple], query: QuerySet):
+        """
+        Apply custom filters to a queryset.
+
+        :param custom_filters: List of named tuples representing custom filters.
```
Comment (dborowiecki) [inline social_protection/custom_filters.py:53]: This information is already provided through in typehint, it's better to give some example of how this named tuple looks like.

File: social_protection/custom_filters.py [hunk 2]
```diff
@@ -44,6 +46,27 @@ def load_definition(self, tuple_type: type, **kwargs) -> List[namedtuple]:
...
+        :type custom_filters: List[namedtuple]
+
+        :param query: The original queryset with filters.
```
Comment (dborowiecki) [inline social_protection/custom_filters.py:56]: Any specific type of QuerySet is required? Like Queryset[BenefitPlan] ?

**Review by dborowiecki — CHANGES_REQUESTED** [review]
Looks solid, I've made some comments regarding documentation.

File: social_protection/custom_filters.py (hunk 2 above)
Comment (sniedzielski) [reply social_protection/custom_filters.py:56]: yes, sth like this. Mostly related to Queryset[Beneficiary] in that case.

File: social_protection/custom_filters.py [hunk 3]
```diff
@@ -44,6 +46,27 @@ def load_definition(self, tuple_type: type, **kwargs) -> List[namedtuple]:
         list_of_tuple_with_definitions = self.__process_schema_and_build_tuple(benefit_plan, tuple_type)
         return list_of_tuple_with_definitions
 
+    def apply_filter_to_queryset(self, custom_filters: List[namedtuple], query: QuerySet):
+        """
+        Apply custom filters to a queryset.
+
+        :param custom_filters: List of named tuples representing custom filters.
+        :type custom_filters: List[namedtuple]
```
[the remark this answers is not in the document]
Comment (sniedzielski) [reply social_protection/custom_filters.py:54]: ok

**Comment by sniedzielski** [comment]
@dborowiecki docsstring fixed

**Review by dborowiecki — APPROVED** [review]

**Pull request merged**